switchbot_ros/scripts/switchbot_ros_server.py

```diff
@@ -24,32 +26,69 @@ def __init__(self):
                 self.token = f.read().replace('\n', '')
         else:
             self.token = token
-        try:
-            self.bots = SwitchBotAPIClient(token=self.token)
-            device_list_str = 'Switchbot device list:\n'
-            device_list = sorted(
-                self.bots.device_list,
-                key=lambda device: str(device['deviceName']))
-            for device in device_list:
-                device_list_str += 'Name: ' + str(device['deviceName'])
-                device_list_str += ', Type: ' + str(device['deviceType'])
-                device_list_str += '\n'
-            rospy.loginfo(device_list_str)
-        except ConnectionError as e: # If the machine is not connected to the internet
-            self.bots = None
-            rospy.logwarn('Failed to connect to the switchbot server. The client would try connecting to it when subscribes the ActionGoal topic.')
+        # Initialize switchbot client
+        self.bots = self.get_switchbot_client()
+        self.print_devices()
         # Actionlib
         self._as = actionlib.SimpleActionServer(
             '~switch', SwitchBotCommandAction,
             execute_cb=self.execute_cb, auto_start=False)
         self._as.start()
+        # Topic
+        self.pub = rospy.Publisher('~devices', DeviceArray, queue_size=1)
+
+    def get_switchbot_client(self):
+
+        try:
+            return SwitchBotAPIClient(token=self.token)
+        except ConnectionError:  # If the machine is not connected to the internet
+            rospy.logwarn('Failed to connect to the switchbot server. The client would try connecting to it when subscribes the ActionGoal topic.')
```

Comment:
This `logwarn` generates so many warn messages when the host not connecting to the internet. Please warn this once when the node is launched but not connecting to the internet. And after connected to the internet, please tells about it with `loginfo`.